### ISSUE TITLE
Integration project: drugst.one

### ISSUE THREAD

**Issue by slobentanzer** [saezlab/BioCypher#85]

**Comment by Maiykol** [comment]
Current state: BioCypher supports neo4j dbms in online and offline mode. The neo4j is thighlty coupled with the rest of BioCypher. 

Motivation: BioCypher should support different databases, in particular PostgreSQL for the integration in Drugst.One. 

Feature implementation: Modularization of the code, in particular separation of the driver and writer class from dbms-specific methods. This allows to write dbms-specific driver and writer classes, implementing only the necessary methods. Verification by implementation of new driver and writer for PostgreSQL and the integration in the Drugst.One project

**Comment by slobentanzer** [comment]
Good plan, fully support the ideas. We should create a general interface (not too fond of inheritance-based approach) for anything that writes to disk. With the driver, I am not sure about other on-line methods and how that would work e.g. with Postgres, but happy to talk. Starting with the disk writing implementation would be easier I guess.

One caveat: @deeenes is currently refactoring BioCypher core, which will probably result in some fairly large changes soon. So I would advise waiting for that update before developing any additional core features.

**Comment by Maiykol** [comment]
Yes, I agree. I started with the disk writing implementation for PostgreSQL as a proof-of-concept and also because we need that for Drugst.One. I made abstract writer and driver classes, which have each ~3 dbms-specific methods that need to be overwritten by a dbms-specific class. We can change that into the driver and the writer returning data and then pasing them to dbms-specific functions that handle the returned data, this way we avoid the inheritance and get the interface.

I didn't start with the online methods, but I think we could implement them to reduce the overhead of writing the data when the database is already set up. However, to this end I would do some performance testing to see if it makes sense and come back to you. 

On the caveat: Okay, I will wait and work for now on the Drugst.One side of the implementation.
